mm2src/mm2_bitcoin/keys/src/address/address_builder.rs

```diff
@@ -50,15 +50,21 @@ impl AddressBuilder {
         self
     }
 
+    /// Sets Address tx output script type as p2pkh or p2wpkh, but also keep the public key stored.
+    pub fn as_pkh_from_pk(mut self, public: &Public) -> Self {
```

Comment:
if 'public' is consumed maybe better to pass it by value?
Also, I think our convention is to name such vars as 'pubkey'